### ISSUE TITLE
Fail to reproduce the CLIPArTT result and the reproduced TENT result is significantly higher than the reported one on Imagenet-C

### ISSUE THREAD

**Issue by ZongfangLiu** [dosowiechi/CLIPArTT#1]
The used dataset is tiny-imangenet-c (200 classes) instead of imagenet-c (1000 classes) which claimed in the paper right?  And the class names used should be full imagenet class names instead of rough one? (Multiple names for one class instead of single name for one class)
There is a bug when loading tiny-imagenet-c, because the teset.classes is only set for imagenet (tiny imagenet 200 in your case) 

<img width="991" alt="Image" src="https://github.com/user-attachments/assets/4f141673-daf3-45f7-a23b-fad539889ae1" />

To reproduce your results, I implemented the dataloader by ourselves (would like to contribute it to your project if you don’t mind) and successfully runned up your codes. However, I can’t reproduce the result of CLIPArTT from Table6 in your paper. We experimented in CIFAR10-C, CIFAR100-C, Tiny Imagenet-C (200 classes) and Imagenet-C (1000 classes). On all the -c datasets, I used the corruption level 5. It shows that the result of CLIPArTT is even worse than zero-shot clip in almost all cases. And the result of TENT is significantly higher than the reported results in the paper.  Please forgive due to time limits I only performed one time validation instead of three, but we think it won’t affect much since the reported variance is minor in the paper. I provide our results comparing clipart and zeroshot clip on cifar100c and comparing tent and zeroshot clip on Imagenet-C (1000class). To my surprise that a harder dataset (1000 class against 200), TENT performs significantly better than the result in your paper.All the experiments are done with the code from your github and the learning rate, batch size are set according to your paper (default in the code), only dataloader is modified since there was a bug in the code. For TENT, due we used learning rate 1e-3 instead of 1e-4 mentioned in the paper due to the default setting in your code.

<img width="925" alt="Image" src="https://github.com/user-attachments/assets/310a3273-7fa0-47d1-aea6-4fdc99f24e56" />

<img width="1274" alt="Image" src="https://github.com/user-attachments/assets/cf42d04c-40f7-48ee-931f-1fceb98311a8" />

<img width="953" alt="Image" src="https://github.com/user-attachments/assets/932bb9d9-006f-49a1-83ec-f5d4eed030ac" />

Is there any potential bugs in the code? Or I missed some settings? Please forgive me if I set anything wrong that leads to such a result. 

Best wishes and looking forward to your reply!

**Comment by GustavoVargasHakim** [comment]
Hi Zongfang, 

Thank you for your inquiry. We are revising the code and will add the dataloader for Imagenet and Imagenet-C. Originally we were going to include TinyImagenet-C, but at the end Imagenet and its variants were more significative. 

In the meantime, we invite you to check this repo (https://github.com/mehrdad-noori/watt) for the paper "WATT: Weight Average Test-Time Adaptation of CLIP", which utilizes a similar loss to CLIPArTT, and where there's a re-implementation of CLIPArTT that's ready to use with Imagenet. The codebase is very straightforward to use.

Thank you again, and we'll soon fix this issue. 

Best, 

Gustavo

**Comment by ZongfangLiu** [comment]
Hi Gustavo,

Thank you for replying in such a short time. I also implemented the dataloader by myself and successfully run the code and reproduced the performance for Zero-shot CLIP. However, I failed to reproduce the result of CLIPArTT as showed in images provided above. And the result for TENT is significantly higher (more than 10% in many corruption types) than the reported one in the paper. 
For the result of CLIPArTT, it should be exactly the same with the setting mentioned in the paper (default by your code I think, maybe I missed some important parameter). 10 round for tying and 1e-3 for the learning rate.
For the result of TENT, the only difference with the paper is I forgot to adjust the learning rate as mentioned in the paper (1e-4) and used 1e-3 learning rate.
I also tried to adjusting learning rate for CLIPArTT, it turns out that if I adjust LR from 1e-3 to 1e-4 or 1e-5, the performance would be almost the same with Zero-shot CLIP which means the model isn't updated at all. And if with 1e-3, then the performance is worse than Zero-shot CLIP.

The idea is really insightful and to me it makes no sense that it would even reduce the performance of Zero-shot CLIP. Is there any other key parameters I need to be care about to reproduce the result of CLIPArTT?  

Thank you for your wonderful work and help!

Best,
Zongfang

**Comment by ZongfangLiu** [comment]
Hi, sorry, I found out what's wrong. I didn't shuffle the loaded dataset, that's why TENT works better while CLIPArTT shows degradation.
